### DISCUSSION TITLE
Get the total number of items from paginated results

### DISCUSSION THREAD

**Discussion by oprypin** [brettcannon/gidgethub#153] [Ideas]
If I'm calling `getiter` on a paginated API, it only provides a lazy iterator over the results. However, I need to know the total number of results in advance, and GitHub provides it in the response - `total_count`, just that the response is completely locked away from the user, unless they call `_make_request` directly like so
https://github.com/brettcannon/gidgethub/blob/3b8941a0ccdad007db09dd601963ce989d64074f/gidgethub/abc.py#L148-L150
(`getitem` also discards that info).

Example of this elsewhere: https://pygithub.readthedocs.io/en/latest/utilities.html?highlight=totalCount#github.PaginatedList.PaginatedList

**Comment by brettcannon** [comment]
How would you expect this to be surfaced in the API?

**Comment by oprypin** [comment]
> Example of this elsewhere: https://pygithub.readthedocs.io/en/latest/utilities.html?highlight=totalCount#github.PaginatedList.PaginatedList

Return an object that has a `total_count` property and also `__aiter__`

**Reply by brettcannon** [reply]
So:
```Python
for total_count, item in gh.getiter(...):
    ...
```
?

**Reply by oprypin** [reply]
```python
it = gh.getiter(...)
count = it.total_count
async for item in it:
    ...
```

**Reply by brettcannon** [reply]
It's an interesting idea. I'm not totally opposed to it, but I have also never had a need to know the total count either.

**Reply by Mariatta** [reply]
I think this is an interesting approach. Currently I have to get the `total_count` by calling the Search API using `getitem`, and if I also need all the search results, then I would make a second call to the Search API using `getiter`.
This solution seems like it could save me a bit of trouble.

**Reply by phlax** [reply]
i would be very plus one to this, just trying to figure out how to migrate some code from `pygithub` that makes use of their `totalCount`

without capturing the data in the request as they do, its looking messy

pygithub count handling:

https://github.com/PyGithub/PyGithub/blob/93b92cd2fce560080dc66aef3b94427623046c5d/github/PaginatedList.py#L143-L168

**Reply by phlax** [reply]
i have an implementation here:

https://github.com/envoyproxy/pytooling/blob/5adf9edecd84bb7e3698c47f62d0706fe96fddc8/aio.api.github/aio/api/github/abstract/iterator.py#L52-L91

i would be happy to fold this upstream, but i guess i need to figure out tests etc
